code/modules/hallucinations/effects/major.dm

```diff
@@ -408,3 +408,140 @@
 	if(images && images[1])
 		images[1].icon = 'icons/mob/alien.dmi'
 		images[1].icon_state = "alienh_pounce"
+
+/obj/effect/hallucination/blob
+	duration = 30 SECONDS
+	/// The blob zombie hallucination.
+	var/obj/effect/hallucination/chaser/attacker/blob_zombie/zombie = null
+	/// The self delusion blob zombie hallucination, triggers on knockdown.
+	var/obj/effect/hallucination/blob_zombify/player_zombie = null
+	/// List of turfs that need expanding from.
+	var/list/turf/expand_queue = list()
+	/// Associative list of turfs that have already been processed.
+	var/list/turf/processed = list()
+	/// The image for the chaser zombie's blob head
+	var/image/chaser_blob_head = null
+	/// The image for the player zombie's blob head
+	var/image/target_blob_head = null
+	/// The delay at which the blob expands in deciseconds. Shouldn't be too low to prevent lag.
+	var/expand_delay = 7.5 SECONDS // Expand 6 times
+	/// Expand timer handle.
+	var/expand_timer = null
+
+/obj/effect/hallucination/blob/Initialize(mapload, mob/living/carbon/target)
+	. = ..()
+	var/list/locs = list()
+	for(var/turf/T in oview(world.view / 2, target))
+		var/light_amount = min(1, T.get_lumcount()) - 0.5
+		if(!is_blocked_turf(T) && light_amount <= 0)
+			locs += T
+	if(!length(locs))
+		qdel(src)
+		return
+	var/turf/T = get_turf(pick(locs))
+	color = pick(	COLOR_BLACK, COLOR_RIPPING_TENDRILS, COLOR_BOILING_OIL, COLOR_ENVENOMED_FILAMENTS, COLOR_LEXORIN_JELLY,
+								COLOR_KINETIC_GELATIN, COLOR_CRYOGENIC_LIQUID, COLOR_SORIUM, COLOR_TESLIUM_PASTE )
+	create_blob(T, core = TRUE)
+	target.playsound_local(T, 'sound/effects/splat.ogg', 50, 1)
+	create_zombie(T)
+	expand_queue += T
+	processed[T] = TRUE
+	expand_timer = addtimer(CALLBACK(src, PROC_REF(expand)), expand_delay, TIMER_LOOP | TIMER_STOPPABLE)
+
+/obj/effect/hallucination/blob/Destroy()
+	deltimer(expand_timer)
+	QDEL_NULL(expand_queue)
+	QDEL_NULL(processed)
```

Comment:
We're just holding onto turfs here, we definitely don't want to delete them but we can empty the list.
Also, I'm pretty sure trying to qdel a list will cause a `bad del`.
```suggestion
	expand_queue.Cut()
	processed.Cut()
```